pkg/disk/nodeserver.go

```diff
@@ -682,9 +680,44 @@ func addDiskXattr(diskID string) (err error) {
 	return unix.Setxattr(device, DiskXattrName, []byte(diskID), 0)
 }
 
-// target format: /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pv-disk-1e7001e0-c54a-11e9-8f89-00163e0e78a0/globalmount
+func ensureUnmounted(mounter k8smount.Interface, target string) error {
+	notmounted, err := mounter.IsLikelyNotMountPoint(target)
+	if err != nil {
+		return fmt.Errorf("failed to check if %s is not a mount point after unmount: %w", target, err)
+	}
+	if !notmounted {
+		return fmt.Errorf("path %s is still mounted after unmount", target)
+	}
+	return nil
+}
+
+func cleanupVolumeDeviceMount(path string) error {
+	err := unix.Unmount(path, 0)
+	if err != nil {
+		switch {
+		case errors.Is(err, unix.ENOENT):
+			return nil
+		case errors.Is(err, unix.EINVAL):
+			// Maybe not mounted, proceed to remove it. If not, unlink will report error.
+		default:
+			return err
+		}
+	}
+
+	errUnlink := unix.Unlink(path)
```

Comment:
i didn't think we have ever link the volumedevice, why do we need to unlink it?

Reply:
`unix.Unlink` is just another way to `os.Remove` a file. I'm using unix package to be consistent with above 'unix.Umount'